### ISSUE TITLE
REQ_WCAG_1.2.2  Captions (Prerecorded)

### ISSUE THREAD

**Issue by ecureuill** [solex2006/SELIProject#10]
# GUIDELINE
## Principle
 **PERCEIVABLE**
> Information and user interface components must be presentable to users in ways they can perceive. This means that users must be able to perceive the information being presented (it can't be invisible to all of their senses) Provide text alternatives for non-text content. Provide captions and other alternatives for multimedia. Create content that can be presented in different ways, including by assistive technologies, without losing meaning. Make it easier for users to see and hear content.

## Level of Conformance 
**A** 
> (the minimum level of conformance), the Web page satisfies all the Level A Success Criteria, or a conforming alternate version is provided.

## General description
Captions are provided for all prerecorded audio content in synchronized media, except when the media is a media alternative for text and is clearly labeled as such.

NOTE: Captions should not be confused with subtitles. Subtitles provide text of only the dialogue and do not include important sounds.

## Intent
>  enable people who are deaf or hard of hearing to watch synchronized media presentations. Captions provide the part of the content available via the audio track. Captions not only include dialogue, but identify who is speaking and include non-speech information conveyed through sound, including meaningful sound effects.

## Examples

1. A captioned tutorial
A video clip shows how to tie a knot. The captions read,
"(music)
Using rope to tie knots was an important skill
for the likes of sailors, soldiers and woodsmen.."
From Sample Transcript Formatting by Whit Anderson.
2. A complex legal document contains synchronized media clips for different paragraphs that show a person speaking the contents of the paragraph. Each clip is associated with its corresponding paragraph. No captions are provided for the synchronized media.
3. An instruction manual containing a description of a part and its necessary orientation is accompanied by a synchronized media clip showing the part in its correct orientation. No captions are provided for the synchronized media clip.
4. An orchestra provides captions for videos of performances. In addition to capturing dialog and lyrics verbatim, captions identify non-vocal music by title, movement, composer, and any information that will help the user comprehend the nature of the audio. For instance captions read,
"[Orchestral Suite No. 3.2 in D major, BWV 1068, Air]
[Johann Sebastian Bach, Composer]
♪ Calm melody with a slow tempo ♪"

## Sufficient Techniques
> Sufficient techniques are reliable ways to meet the success criteria.
> - From an author's perspective: If you use the sufficient techniques for a given criterion correctly and it is accessibility-supported for your users, you can be confident that you met the success criterion.
> - From an evaluator's perspective: If web content implements the sufficient techniques for a given criterion correctly and it is accessibility-supported for the content's users, it conforms to that success criterion. (The converse is not true; if content does not implement these sufficient techniques, it does not necessarily fail the success criteria, as explained in Testing Techniques below.)
> There may be other ways to meet success criteria besides the sufficient techniques in W3C's Techniques for WCAG document, as Other Techniques below. (See also Techniques are Informative above.)

- G93
- G87

### [G93: Providing open (always visible) captions](https://www.w3.org/WAI/WCAG21/Techniques/general/G93)
> The objective of this technique is to provide a way for people who are deaf or otherwise have trouble hearing the dialogue in audio visual material to be able to view the material. With this technique all of the dialogue and important sounds are embedded as text in the video track. As a result they are always visible and no special support for captioning is required by the user agent.

**Example**
- In order to ensure that everyone can view their online movies, even if users do not know how to turn on captions in their media player, a library association puts the captions directly into the video.
- A news organization provides open captions on all of its material

### Test Procedure

1. Watch the synchronized media with closed captioning turned off.
2. Check that captions (of all dialogue and important sounds) are visible.

**Expected Results**

#2 is true

### [G87: Providing closed captions](https://www.w3.org/WAI/WCAG21/Techniques/general/G87) using any readily available media format that has a video player that supports closed captioning
> The objective of this technique is to provide a way for people who have hearing impairments or otherwise have trouble hearing the dialogue in synchronized media material to be able to view the material and see the dialogue and sounds - without requiring people who are not deaf to watch the captions. With this technique all of the dialogue and important sounds are embedded as text in a fashion that causes the text not to be visible unless the user requests it. As a result they are visible only when needed. This requires special support for captioning in the user agent.

**Example**

1. In order to ensure that users who are deaf can use their interactive educational materials, the college provides captions and instructions for turning on captions for all of their audio interactive educational programs.
2. The online movies at a media outlet all include captions and are provided in a format that allows embedding of closed captions.
3. Special caption files including synchronization information are provided for an existing movie. Players are available that can play the captions in a separate window on screen, synchronized with the movie window.
4. A video of a local news event has captions provided that can be played over the video or in a separate window depending on the player used.

### Test Procedure

1. Turn on the closed caption feature of the media player
2. View the synchronized media content
3. Check that captions (of all dialogue and important sounds) are visible

**Expected Results**

#3 is true

### [G87: Providing closed captions](https://www.w3.org/WAI/WCAG21/Techniques/general/G87) using any of the technology-specific techniques below (I'm just refering to h95 technique, for others see full documentation)
  - [H95: Using the track element to provide captions](https://www.w3.org/WAI/WCAG21/Techniques/html/H95)
  > The objective of this technique is to use the HTML5 track element to specify a captions timed text track for a video element. Caption timed text tracks contain transcription or translation of the dialogue, sound effects, relevant musical cues, and other relevant audio information, suitable for when sound is unavailable or not clearly audible.
**Example**
A video element for a video in the English language with an English caption track. The captions are provided in the WebVTT format.
```
<video poster="myvideo.png" controls>
  <source src="myvideo.mp4" srclang="en" type="video/mp4">
  <source src="myvideo.webm" srclang="fr" type="video/webm">
  <track src="myvideo_en.vtt" kind="captions" srclang="en" label="English">
  <track src="myvideo_fr.ttml" kind="captions" srclang="fr" label="French">
</video>      
```
### Test Procedure

1. Check that the video contains a track element of kind captions in the language of the video.

**Expected Results**

Check #1 is true.

# SELI CONTEXT
> How to implement this requirement on SELI Platform?

```
<!-- SEIZURE (if risk is true ) -->
<input type="checkbox" value="aware" aria-expanded="false" aria-controls="id_video_box">This video is hidden because has seizure risks, choose this option to show the video</button>
<div id="id_video_box">
   <h3>Video title</h3>
   <p id="id_short_desc"><!-- SHORT DESCRIPTION --></p>
   <h3><button aria-expanded="false" aria-controls="id_req_acc">Available accessibility resources</button></h3>
   <ul id="id_req_acc">
  	<li>Audio description</li>
  	<li>Closed captions</li>
 	<li>Transcriptions</li>
  	<li>Sign Language</li>
   </ul>
   <video id=video poster="poster.png" autoplay="false" aria-label="id_short_desc" aria-describedby="id_long_desc">
       <source id='mp4' src="trailer.mp4" type='video/mp4'/>
       <!-- Track to be used for CAPTIONS -->
       <track kind="captions" label="English[cc]" src="captions_en.vtt" srclang="en" default></track>
       <!-- Track to be used for AUDIO-DESCRIPTION as .vtt -->
       <track kind="descriptions" label="English[desc]" src="audiodesc_de.vtt" srclang="en"></track>
       <p>Your user agent does not support the HTML5 Video element.</p>
   </video>
</div>

<button aria-expanded="false" aria-controls="id_long_desc">Video Transcription</button>
<div><p  id="id_long_desc"><!-- TRANSCRIPTIONS --></p></div>

<div> <!-- AUDIO-DESCRIPTION as audio file -->
  <p id="id_audio_short_desc"><!-- Audio description of Video Title --></p>
  <audio id=audio autoplay="false" aria-labelby="id_audio_short_desc">
    <source src="audio.mp3" type='audio/mpeg'/>
  </audio>
</div>
```
## Tasks
List the tasks required to accomplish the requirement (not exaustive)

## System's contents
> All contents not created by users (e.g. seli "about" section text, buttons, fields) that STUDENT could consume 
- [ ] **DEVELOPER** MUST ...
- [ ] **TEACHER** MUST have the option to ...
- [ ] **STUDENT** must render teacher's content ...

# :notebook_with_decorative_cover:   References

- [How to Meet WCAG (Quick Reference)](https://www.w3.org/WAI/WCAG21/quickref/#audio-only-and-video-only-prerecorded)
- [Understanding Techniques for WCAG Success Criteria](https://www.w3.org/WAI/WCAG21/Understanding/understanding-techniques)

# GLOSSARY 
:warning: The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119/8174](http://www.rfc-editor.org/info/rfc8174).
> 1. **MUST** This word, or the terms "REQUIRED" or "SHALL", mean that the definition is an absolute requirement of the specification.
> 2. **MUST NOT** This phrase, or the phrase "SHALL NOT", mean that the definition is an absolute prohibition of the specification.
> 3. **SHOULD** This word, or the adjective "RECOMMENDED", mean that there may exist valid reasons in particular circumstances to ignore a particular item, but the full implications must be understood and carefully weighed before choosing a different course.
> 4. **SHOULD NOT** This phrase, or the phrase "NOT RECOMMENDED" mean that there may exist valid reasons in particular circumstances when the particular behavior is acceptable or even useful, but the full implications should be understood and the case carefully weighed before implementing any behavior described with this label.
> 5. **MAY**   This word, or the adjective "OPTIONAL", mean that an item is truly optional. An implementation which does not include a particular option MUST be prepared to interoperate with another implementation which does include the option, though perhaps with reduced functionality. In the same vein an implementation which does include a particular option MUST be prepared to interoperate with another implementation which does not include the option (except, of course, for the feature the option provides.)
